osu.Game.Tests/Visual/Editing/TestSceneTimelineHitObjectBlueprint.cs

```diff
@@ -111,5 +117,148 @@ public void TestDisallowRepeatsOnZeroDurationObjects()
+
+            AddStep("add circle", () =>
+            {
+                EditorBeatmap.Clear();
+                EditorBeatmap.Add(new HitCircle
+                {
+                    Position = new Vector2(256, 256),
+                    StartTime = 2700,
+                    Samples =
+                    {
+                        new HitSampleInfo(HitSampleInfo.HIT_NORMAL)
+                    }
+                });
+            });
+
+            AddStep("open hitsound popover", () =>
+            {
+                samplePointPiece = this.ChildrenOfType<SamplePointPiece>().Single();
+                InputManager.MoveMouseTo(samplePointPiece);
+                InputManager.PressButton(MouseButton.Left);
+                InputManager.ReleaseButton(MouseButton.Left);
+            });
+
+            AddStep("add whistle addition", () =>
+            {
+                popover = this.ChildrenOfType<SamplePointPiece.SampleEditPopover>().First();
+                var whistleTernaryButton = popover.ChildrenOfType<DrawableTernaryButton>().First();
+                InputManager.MoveMouseTo(whistleTernaryButton);
+                InputManager.PressButton(MouseButton.Left);
+                InputManager.ReleaseButton(MouseButton.Left);
+            });
+
+            AddAssert("has whistle sample", () => EditorBeatmap.HitObjects.First().Samples.Any(o => o.Name == HitSampleInfo.HIT_WHISTLE));
+
+            AddStep("change bank name", () =>
+            {
+                var bankTextBox = popover.ChildrenOfType<LabelledTextBox>().First();
+                bankTextBox.Current.Value = "soft";
+            });
+
+            AddAssert("bank name changed", () =>
+                EditorBeatmap.HitObjects.First().Samples.Where(o => o.Name == HitSampleInfo.HIT_NORMAL).All(o => o.Bank == "soft")
+                && EditorBeatmap.HitObjects.First().Samples.Where(o => o.Name != HitSampleInfo.HIT_NORMAL).All(o => o.Bank == "normal"));
+
+            AddStep("change addition bank name", () =>
+            {
+                var bankTextBox = popover.ChildrenOfType<LabelledTextBox>().ToArray()[1];
+                bankTextBox.Current.Value = "drum";
+            });
+
+            AddAssert("addition bank name changed", () =>
+                EditorBeatmap.HitObjects.First().Samples.Where(o => o.Name == HitSampleInfo.HIT_NORMAL).All(o => o.Bank == "soft")
+                && EditorBeatmap.HitObjects.First().Samples.Where(o => o.Name != HitSampleInfo.HIT_NORMAL).All(o => o.Bank == "drum"));
+
+            AddStep("change volume", () =>
+            {
+                var bankTextBox = popover.ChildrenOfType<IndeterminateSliderWithTextBoxInput<int>>().Single();
+                bankTextBox.Current.Value = 30;
+            });
+
+            AddAssert("volume changed", () => EditorBeatmap.HitObjects.First().Samples.All(o => o.Volume == 30));
+        }
+
+        [Test]
+        public void TestNodeSamplePointPiece()
```

Comment:
These tests need to clear the popover at the start or the end, currently running them more than once results in a failure.

https://github.com/ppy/osu/assets/191335/1ee0af08-bcf8-4809-bb10-c0fffbdc482f